lib/miq_expression.rb

```diff
@@ -156,10 +156,15 @@ def self._to_human(exp, options = {})
   end
 
   def to_ruby(tz = nil)
-    return "" unless valid?
-    tz ||= "UTC"
-    @ruby ||= self.class._to_ruby(exp.deep_clone, context_type, tz)
-    @ruby.dup
+    # TODO: we cache the ruby expression regardless if the tz is different
+    if @ruby
+      @ruby.dup
+    elsif valid?
+      @ruby ||= self.class._to_ruby(exp.deep_clone, context_type, tz || "UTC".freeze)
+      @ruby.dup
+    else
+      ""
```

Comment:
I was more concerned with a change in behavior.  Perhaps it would never be invalid and made valid but it was a change that I don't think we needed for the issue we're solving here.